# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/fkashaf19-afk/ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

**Lane: Lane 2 — Refresh / Content Opportunity Scoring**

Why this lane:
- The starter pipeline in this repo (scripts/01-05) is already built for this exact lane,
  so I can validate early whether the approach has signal before committing 7 weeks to it.
- It maps to a real, recurring decision — which pages to review first — instead of an
  abstract prediction task.
- The lane guide already reports verified results from this pipeline, giving me a
  baseline to compare against from day one.

A simple rule could catch the obvious cases, but the starter pipeline shows a
learned model finds much more signal than a fixed rule can (precision@50 jumps
from 0.240 to 0.740) — that gap is exactly why ML earns its place here, not
just a hand-written if-statement.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Decision:** Given a limited review budget (a content team can realistically review
~50 pages a week), which pages should be reviewed first — for refresh, expansion, or
pruning?

**Unit of analysis:** one content page (a `content_id`), scored and ranked
individually — not a client or a time-period aggregate.

**Who acts, and how:** A content editor opens the ranked queue and works top-down,
applying the suggested action tied to each page's reason code (e.g. rewrite
title/meta, refresh copy, monitor).

**Cost of a wrong call:**
- False positive (reviewed a page that wasn't a real problem): low cost — wasted
  editor time, deprioritized effort.
- False negative (missed a page that was genuinely declining with real demand):
  higher cost — lost traffic that compounds the longer it's ignored.

This asymmetry is why precision@K (not plain accuracy) is the right metric — it
matches how a reviewer actually works down a ranked list.

Load the starter CSV and pull real numbers to check this lane is worth pursuing.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [8]:
import os

if os.path.basename(os.getcwd()) != "ml-internship":
    if not os.path.exists("ml-internship"):
        !git clone https://github.com/fkashaf19-afk/ml-internship.git
    %cd ml-internship

In [9]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total rows:", len(df))

# is_declining_label isn't a raw column — it's derived from trend_direction,
# exactly as the starter pipeline (01_prepare_features.py) computes it.
df['is_declining_label'] = df['trend_direction'] == 'down'

print("Rows labeled declining:", df['is_declining_label'].sum())
print("Share declining:", round(df['is_declining_label'].mean(), 3))

Total rows: 30000
Rows labeled declining: 16262
Share declining: 0.542


**Numbers supporting this lane:**

1. Of the 30,000 starter rows, 54.2% (16,262 rows) are labeled declining under the
   current proxy definition — enough volume for a ranking approach to have real
   material to work with, not a handful of edge cases.

2. The starter pipeline (already executed, see outputs/model_report.md) shows a
   learned ranking clearly beats a fixed rule:
   - Baseline rules: precision@50 = 0.240 (~12 of top 50 correct)
   - Random forest: precision@50 = 0.740 (~37 of top 50 correct)

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

**What I can claim:**
- Observed patterns associated with decline or opportunity, based on measured signals.
- Decision-support recommendations that help a human reviewer prioritize limited time.

**What I cannot claim:**
- That a refresh will *cause* recovery — that requires a controlled experiment I'm
  not running here.
- That I've discovered a Google ranking factor.
- That "declining" is ground truth — it's a proxy label I chose, not an objective fact.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.